.github/workflows/build.yml

```diff
@@ -0,0 +1,37 @@
+name: Nix Flake Build
+
+on:
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest  # Specify self-hosted and NixOS labels
+    steps:
+    - uses: actions/checkout@v4
+      
+    - uses: cachix/install-nix-action@v27
+      with:
+        github_access_token: ${{ secrets.GITHUB_TOKEN }}
+    
+    - uses: cachix/cachix-action@v15
+      with:
+        name: utcsheffield
+        authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+
+    - name: Build makerlab-3040
+      run: nix build nixpkgs#nixos-rebuild && ./result/bin/nixos-rebuild build --flake .#makerlab-3040
```

Comment:
Also, append --dry-run to that build command, so we don't actually build it every time.
We definitely only need to build it once

Reply:
We should also explore using a matrix for the build command, as it's easier to maintain

Reply:
you try that i've never done that
on the bright side it works and i think we get unlimited workflow minutes because the repo is public

Reply:
> Also, append --dry-run to that build command, so we don't actually build it every time. We definitely only need to build it once

To what build command, also building nixos-rebuild separately then running each build command only works for the first one...??